archive/multisource.go

```diff
@@ -0,0 +1,181 @@
+package archive
+
+import (
+	"context"
+	"io"
+
+	"github.com/brimsec/zq/ast"
+	"github.com/brimsec/zq/driver"
+	"github.com/brimsec/zq/filter"
+	"github.com/brimsec/zq/pkg/iosrc"
+	"github.com/brimsec/zq/pkg/nano"
+	"github.com/brimsec/zq/scanner"
+	"github.com/brimsec/zq/zbuf"
+	"github.com/brimsec/zq/zio"
+	"github.com/brimsec/zq/zio/detector"
+	"github.com/brimsec/zq/zio/zngio"
+	"github.com/brimsec/zq/zng/resolver"
+)
+
+// A SpanInfo is a logical view of the records within a time span, stored
+// in one or more Chunks.
+type SpanInfo struct {
+	First nano.Ts // timestamp of first record in this span
+	Last  nano.Ts // timestamp of last record in this span
```

Comment:
I could: Chunk has 'First,Last' because it felt natural to use the first and last record's timestamps as part of a file's name, and because that also communicated the ordering of the records in the file. But that's not true for SpanInfo, its time is always just used as a filter.